packages/asset-swapper/contracts/src/WooPPSampler.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.6;
+pragma experimental ABIEncoderV2;
+import "./interfaces/IWooPP.sol";
+import "./SamplerUtils.sol";
+import "./ApproximateBuys.sol";
+
+
+contract WooPPSampler is SamplerUtils, ApproximateBuys{
+
+   function query(
+        uint amountIn, 
+        address tokenIn, 
+        address tokenOut,
+        address pool
+    ) internal view returns (uint256 amountOut) {
+        if (amountIn == 0) { 
+            return 0; 
+        }
+        address quoteToken = IWooPP(pool).quoteToken();
+        if (tokenIn == quoteToken) {
+            amountOut = IWooPP(pool).querySellQuote(tokenOut, amountIn);
+        } else if (tokenOut == quoteToken) {
+            amountOut = IWooPP(pool).querySellBase(tokenIn, amountIn);
+        } else {
+            uint quoteAmount = IWooPP(pool).querySellBase(tokenIn, amountIn);
+            amountOut = IWooPP(pool).querySellQuote(tokenOut, quoteAmount);
+        }
+    }
+
+    /// @dev Sample sell quotes from WooFI.
+    /// @param pool Address of the pool we are sampling from
+    /// @param takerToken Address of the taker token (what to sell).
+    /// @param makerToken Address of the maker token (what to buy).
+    /// @param takerTokenAmounts Taker token sell amount for each sample (sorted in ascending order).
+    /// @return makerTokenAmounts Maker amounts bought at each taker token
+    ///         amount.
+    
+    function sampleSellsFromWooPP(
+        address pool,
+        address takerToken,
+        address makerToken,
+        uint256[] memory takerTokenAmounts
+    )
+        public
+        view
+        returns (uint256[] memory makerTokenAmounts)
+        {
```

Comment:
there are multiple left over comments (from debugging?) in this block. 

nit/optional: I would also remove comments that don't provide much value. For example `Use WooFI's query function for sampling` and `Break if there are 0 amounts` don't provide much value in my opinion.